source/postprocess/visualization/velocity_residual.cc

```diff
@@ -0,0 +1,169 @@
+			{
+		      for (unsigned int d = 0; d < dim; ++d)
+		    	velocity[d] = Utilities::AsciiDataBoundary<dim>::get_data_component(
+		    	  							this->get_geometry_model().translate_symbolic_boundary_name_to_id("top"),
+		    	  							input_data.evaluation_points[q], d);
+		      velocity = Utilities::Coordinates::spherical_to_cartesian_vector(velocity, input_data.evaluation_points[q]);
+
+		      for (unsigned int d = 0; d < dim; ++d)
+		        computed_quantities[q](d) = (velocity[d] - input_data.solution_values[q][d] * velocity_scaling_factor) ;
+			}
+		    else
+		    {
+		      for (unsigned int d = 0; d < dim; ++d)
+		      {
+			  velocity[d] = Utilities::AsciiDataBoundary<dim>::get_data_component(
+													  this->get_geometry_model().translate_symbolic_boundary_name_to_id("top"),
+													  input_data.evaluation_points[q], d);
+			  computed_quantities[q](d) = (velocity[d] - input_data.solution_values[q][d] * velocity_scaling_factor) ;
+
+		      }
+		    }
+		  }
+		}
+      }
+
+
+      template <int dim>
+      void
+	  VelocityResidual<dim>::declare_parameters (ParameterHandler &prm)
+	  {
+    	prm.enter_subsection("Postprocess");
+    	{
+          prm.enter_subsection("Visualization");
+    	  {
+		    Utilities::AsciiDataBase<dim>::declare_parameters(prm,
+						  "$ASPECT_SOURCE_DIR/data/initial-temperature/adiabatic-boundary/",
+						  "adiabatic_boundary.txt", "Surface properties");
+		    prm.declare_entry ("Use spherical unit vectors", "false",
+		    Patterns::Bool (),
+			"Specify velocity as r, phi, and theta components "
+			"instead of x, y, and z. Positive velocities point up, east, "
+			"and north (in 3D) or out and clockwise (in 2D). "
+			"This setting only makes sense for spherical geometries."
+		    );
+    	  }
+    	  prm.leave_subsection();
+    	  }
+    	prm.leave_subsection();
+	  }
+
+
+      template <int dim>
+      void
+	  VelocityResidual<dim>::parse_parameters (ParameterHandler &prm)
+	  {
+	    prm.enter_subsection("Postprocess");
+		  prm.enter_subsection("Visualization");
+		  {
+			Utilities::AsciiDataBase<dim>::parse_parameters(prm, "Surface properties");
+			use_spherical_unit_vectors = prm.get_bool("Use spherical unit vectors");
+			if (use_spherical_unit_vectors)
+			  AssertThrow (this->get_geometry_model().natural_coordinate_system() == Utilities::Coordinates::spherical,
+			   ExcMessage ("Spherical unit vectors should not be used "
+				"when geometry model is not spherical."));
+		  }
+		  prm.leave_subsection();
+	    prm.leave_subsection();
+	  }
+
+
+    }
+  }
+}
+
+
+// explicit instantiations
+namespace aspect
+{
+  namespace Postprocess
+  {
+    namespace VisualizationPostprocessors
+    {
+      ASPECT_REGISTER_VISUALIZATION_POSTPROCESSOR(VelocityResidual,
+                                                  "velocity residual",
+                                                  "A visualization for caculating the surface velocity to compare"
```

Comment:
Please extend the documentation a bit, mentioning that this outputs the difference between the surface velocities and some reference velocity model.